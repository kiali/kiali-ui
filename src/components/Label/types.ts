export interface Labels {
  name: string;
  value: string;
}
