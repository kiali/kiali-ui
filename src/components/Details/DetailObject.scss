ul.details {
  -webkit-padding-start: 0em;
  padding-left: 0.5em;
  overflow-wrap: break-word;
  list-style-type: none;
}
