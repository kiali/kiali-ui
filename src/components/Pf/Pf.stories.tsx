import '../../app/App.css';

import * as React from 'react';
import { storiesOf } from '@storybook/react';

import PFHeader from './PfHeader';
import PFContainerNavVertical from './PfContainerNavVertical';

const stories = storiesOf('Pf', module);

stories.add('PFContainerNavVertical', () => (
  <PFContainerNavVertical>
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
      Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </div>
  </PFContainerNavVertical>
));

stories.add('PFHeader', () => (
  <PFContainerNavVertical>
    <PFHeader>
      <h2> Awesome header </h2>
    </PFHeader>
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
      Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </div>
  </PFContainerNavVertical>
));
