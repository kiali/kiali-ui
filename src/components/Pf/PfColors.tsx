export enum PfColors {
  Black100 = '#fafafa',
  Black150 = '#f5f5f5',
  Black200 = '#ededed',
  Black300 = '#d1d1d1',
  Black400 = '#bbb',
  Black500 = '#8b8d8f',
  Black600 = '#72767b',
  Black700 = '#4d5258',
  Black800 = '#393f44',
  Black900 = '#292e34',
  Black1000 = '#030303',
  Blue50 = '#def3ff',
  Blue100 = '#bee1f4',
  Blue200 = '#7dc3e8',
  Blue300 = '#39a5dc',
  Blue400 = '#0088ce',
  Blue500 = '#00659c',
  Blue600 = '#004368',
  Blue700 = '#002235',
  Cyan100 = '#bedee1',
  Cyan200 = '#7dbdc3',
  Cyan300 = '#3a9ca6',
  Cyan400 = '#007a87',
  Cyan500 = '#005c66',
  Cyan600 = '#003d44',
  Cyan700 = '#001f22',
  Gold100 = '#fbeabc',
  Gold200 = '#f9d67a',
  Gold300 = '#f5c12e',
  Gold400 = '#f0ab00',
  Gold500 = '#b58100',
  Gold600 = '#795600',
  Gold700 = '#3d2c00',
  Green100 = '#cfe7cd',
  Green200 = '#9ecf99',
  Green300 = '#6ec664',
  Green400 = '#3f9c35',
  Green500 = '#2d7623',
  Green600 = '#1e4f18',
  Green700 = '#0f280d',
  LightBlue100 = '#beedf9',
  LightBlue200 = '#7cdbf3',
  LightBlue300 = '#35caed',
  LightBlue400 = '#00b9e4',
  LightBlue500 = '#008bad',
  LightBlue600 = '#005c73',
  LightBlue700 = '#002d39',
  LightGreen100 = '#e4f5bc',
  LightGreen200 = '#c8eb79',
  LightGreen300 = '#ace12e',
  LightGreen400 = '#92d400',
  LightGreen500 = '#6ca100',
  LightGreen600 = '#486b00',
  LightGreen700 = '#253600',
  Orange100 = '#fbdebf',
  Orange200 = '#f7bd7f',
  Orange300 = '#f39d3c',
  Orange400 = '#ec7a08',
  Orange500 = '#b35c00',
  Orange600 = '#773d00',
  Orange700 = '#3b1f00',
  Purple100 = '#c7bfff',
  Purple200 = '#a18fff',
  Purple300 = '#8461f7',
  Purple400 = '#703fec',
  Purple500 = '#582fc0',
  Purple600 = '#40199a',
  Purple700 = '#1f0066',
  Red100 = '#cc0000',
  Red200 = '#a30000',
  Red300 = '#8b0000',
  Red400 = '#470000',
  Red500 = '#2c0000',

  White = '#fff',
  Black = '#030303',

  Blue = '#0088ce', // Blue400
  Cyan = '#007a87', // Cyan400
  Gold = '#f0ab00', // Gold400
  Green = '#3f9c35', // Green400
  LightBlue = '#00b9e4', // LightBlue400
  LightGreen = '#92d400', // LightGreen400
  Orange = '#ec7a08', // Orange400
  Red = '#cc0000', // Red100

  // Kiali colors that use PF colors
  Gray = '#72767b' // Black600
}

export enum PFColorVars {
  DangerColor = 'var(--pf-global--danger-color--100)',
  WarningColor = 'var(--pf-global--warning-color--100)',
  SuccessColor = 'var(--pf-global--success-color--100)'
}
