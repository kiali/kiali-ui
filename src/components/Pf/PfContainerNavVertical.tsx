import React from 'react';

const PfContainerNavVertical = props => {
  return <div className="container-fluid container-pf-nav-pf-vertical">{props.children}</div>;
};

export default PfContainerNavVertical;
