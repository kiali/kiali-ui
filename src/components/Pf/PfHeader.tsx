import React from 'react';

type PfHeaderProps = {
  // none yet
};

const PfHeader: React.SFC<PfHeaderProps> = props => {
  return <div className="page-header">{props.children}</div>;
};

export default PfHeader;
