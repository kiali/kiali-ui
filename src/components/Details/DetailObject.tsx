import * as React from 'react';
import Label from '../Label/Label';
import { Icon } from 'patternfly-react';
import './DetailObject.css';

interface DetailObjectProps {
  name: string;
  detail: any;
  labels?: string[];
  exclude?: string[];
  validation?: Validation;
}

interface Validation {
  message: string;
  icon: string;
  color: string;
}

class DetailObject extends React.Component<DetailObjectProps> {
  // Pseudo unique ID generator used for keys
  // The recursive nature of buildList() requires uniques list keys.
  // Modified from https://gist.github.com/gordonbrander/2230317
  generateKey() {
    return (
      'key_' +
      Math.random()
        .toString(36)
        .substr(2, 9)
    );
  }

  label(key: string, value: string) {
    return <Label name={key} value={value} />;
  }

  checkLabel(name: string) {
    if (!this.props.labels) {
      return false;
    }
    return this.props.labels.indexOf(name) > -1;
  }

  canDisplay(name: string) {
    return this.props.exclude == null || !this.props.exclude.includes(name);
  }

  // buildList returns a recursive list of all items within value. It shows a validation
  // only for the first iteration (when depth is 0)
  buildList(name: string, value: any, isLabel: boolean, depth: number): any {
    if (!this.canDisplay(name)) {
      return '';
    }

    const valueType = typeof value;
    if (valueType === 'string' || valueType === 'number' || valueType === 'boolean') {
      return (
        <div className="label-collection">
          {isLabel ? (
            this.label(name, value)
          ) : (
            <span>
              <span className="text-capitalize">{name ? `[${name}]` : name}</span> {value.toString()}
            </span>
          )}
        </div>
      );
    }

    const childrenList: any[] = [];
    const listKey = this.generateKey();
    const checkLabel = this.checkLabel(name);
    if (Array.isArray(value)) {
      value.forEach((v, i) => {
        const vType = typeof v;
        if (vType === 'string' || vType === 'number' || vType === 'boolean') {
          childrenList.push(<li key={listKey + '_i' + i}>{v}</li>);
        } else {
          Object.keys(v).forEach((key, j) => {
            const childList = this.buildList(key, v[key], checkLabel, depth + 1);
            childrenList.push(<li key={listKey + '_i' + i + '_j' + j}>{childList}</li>);
          });
        }
      });
    } else {
      Object.keys(value || {}).forEach((key, k) => {
        const childList = this.buildList(key, value[key], checkLabel, depth + 1);
        childrenList.push(<li key={listKey + '_k' + k}>{childList}</li>);
      });
    }

    return childrenList.length > 0 ? (
      <div>
        <strong className="text-capitalize">{name}</strong>
        {depth === 0 && !!this.props.validation && this.props.validation.message ? (
          <div>
            <p style={{ color: this.props.validation.color }}>
              <Icon type="pf" name={this.props.validation.icon} /> {this.props.validation.message}
            </p>
          </div>
        ) : (
          undefined
        )}
        <ul className={'details'}>{childrenList}</ul>
      </div>
    ) : (
      ''
    );
  }

  render() {
    const findLabels = typeof this.props.labels !== 'undefined' && this.props.labels.length > 0;

    const objectList = this.buildList(this.props.name, this.props.detail, findLabels, 0);
    return <div>{objectList}</div>;
  }
}

export default DetailObject;
