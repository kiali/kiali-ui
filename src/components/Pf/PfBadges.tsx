import { Badge, Tooltip, TooltipPosition } from '@patternfly/react-core';
import React, { CSSProperties } from 'react';
import { style } from 'typestyle';
import { PFColors } from './PfColors';

export type PFBadgeType = {
  badge: string;
  tt?: React.ReactFragment;
};

// PF Badges used by Kiali, keep alphabetized
// avoid duplicate badge letters, especially if they may appear on the same page
export const PFBadges = Object.freeze({
  App: { badge: 'A', tt: 'Application' } as PFBadgeType,
  Adapter: { badge: 'A', tt: 'Adapter' } as PFBadgeType,
  AttributeManifest: { badge: 'AM', tt: 'Attribute Manifest' } as PFBadgeType,
  AuthorizationPolicy: { badge: 'AP', tt: 'Authorization Policy' } as PFBadgeType,
  Cluster: { badge: 'CL', tt: 'Cluster' } as PFBadgeType,
  ClusterRBACConfig: { badge: 'CRC', tt: 'Cluster RBAC Configuration' } as PFBadgeType,
  Container: { badge: 'C', tt: 'Container' } as PFBadgeType,
  DestinationRule: { badge: 'DR', tt: 'Destination Rule' } as PFBadgeType,
  EnvoyFilter: { badge: 'EF', tt: 'Envoy Filter' } as PFBadgeType,
  ExternalService: { badge: 'ES', tt: 'External Service' } as PFBadgeType,
  FederatedService: { badge: 'FS', tt: 'Federated Service' } as PFBadgeType,
  Gateway: { badge: 'G', tt: 'Gateway' } as PFBadgeType,
  Handler: { badge: 'H', tt: 'Handler' },
  Host: { badge: 'H', tt: 'Host' },
  Instance: { badge: 'I', tt: 'Instance' },
  MeshPolicy: { badge: 'MP', tt: 'Mesh Policy' } as PFBadgeType,
  Namespace: { badge: 'NS', tt: 'Namespace' } as PFBadgeType,
  Operation: { badge: 'O', tt: 'Operation' } as PFBadgeType,
  PeerAuthentication: { badge: 'PA', tt: 'Peer Authentication' } as PFBadgeType,
  Pod: { badge: 'P', tt: 'Pod' } as PFBadgeType,
  Policy: { badge: 'P', tt: 'Policy' } as PFBadgeType,
  RBACConfig: { badge: 'RC', tt: 'RBAC Configuration' } as PFBadgeType,
  RequestAuthentication: { badge: 'RA', tt: 'Request Authentication' } as PFBadgeType,
  Rule: { badge: 'R', tt: 'Rule' } as PFBadgeType,
  Service: { badge: 'S', tt: 'Service' } as PFBadgeType,
  ServiceEntry: { badge: 'SE', tt: 'Service Entry' } as PFBadgeType,
  ServiceRole: { badge: 'SR', tt: 'Service Role' } as PFBadgeType,
  ServiceRoleBinding: { badge: 'SRB', tt: 'Service Role Binding' } as PFBadgeType,
  Sidecar: { badge: 'SC', tt: 'Istio Sidecar Proxy' } as PFBadgeType,
  Template: { badge: 'T', tt: 'Template' } as PFBadgeType,
  Unknown: { badge: 'U', tt: 'Unknown' } as PFBadgeType,
  VirtualService: { badge: 'VS', tt: 'Virtual Service' } as PFBadgeType,
  Workload: { badge: 'W', tt: 'Workload' } as PFBadgeType,
  WorkloadEntry: { badge: 'WE', tt: 'Workload Entry' } as PFBadgeType,
  WorkloadGroup: { badge: 'WG', tt: 'Workload Group' } as PFBadgeType
});

export const kialiBadge = style({
  backgroundColor: PFColors.Badge,
  borderRadius: '30em',
  marginRight: '10px'
});

type PFBadgeProps = {
  badge: PFBadgeType;
  className?: string; // default=kialiBadge
  id?: string;
  isRead?: boolean;
  key?: string;
  position?: TooltipPosition; // default=auto
  style?: CSSProperties;
};

export class PFBadge extends React.PureComponent<PFBadgeProps> {
  render() {
    const key = `pfbadge-${this.props.badge.badge}`;
    const badge = (
      <Badge
        className={this.props.className || kialiBadge}
        id={this.props.id || key}
        isRead={this.props.isRead || false}
        key={this.props.key || key}
        style={this.props.style}
      >
        {this.props.badge.badge}
      </Badge>
    );

    return !this.props.badge.tt ? (
      badge
    ) : (
      <Tooltip
        content={<>{this.props.badge.tt}</>}
        id={`tt-${this.props.id || key}`}
        key={`tt-${this.props.key || key}`}
        position={this.props.position || TooltipPosition.auto}
      >
        {badge}
      </Tooltip>
    );
  }
}

export const getPFBadge = (badge: string, tt?: React.ReactFragment): React.ReactFragment => {
  return <PFBadge badge={{ badge: badge, tt: tt }} />;
};
