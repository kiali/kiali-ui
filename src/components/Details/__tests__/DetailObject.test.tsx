import * as React from 'react';
import { shallow } from 'enzyme';
import { Icon } from 'patternfly-react';
import { default as DetailObject } from '../DetailObject';
import { PfColors } from '../../Pf/PfColors';
import { DestinationWeight } from '../../../types/IstioObjects';
import { shallowToJson } from 'enzyme-to-json';

describe('DetailObject test', () => {
  const detail: DestinationWeight = {
    destination: {
      host: 'reviews',
      subset: 'v1',
      port: {
        number: 22,
        name: 'ssh'
      }
    },
    weight: 85
  };

  const mockRandom = () => {
    const mockMath = Object.create(global.Math);
    mockMath.random = () => 0.8;
    global.Math = mockMath;
  };

  it('prints a nested list with all attributes in the detail', () => {
    mockRandom();

    const wrapper = shallow(<DetailObject name={name} detail={detail} />);

    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    expect(wrapper.html()).toContain('<span class="text-capitalize">[host]</span>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[subset]</span>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[weight]</span>');

    expect(wrapper.html()).toContain('<strong class="text-capitalize">port</strong>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[number]</span>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[name]</span>');
  });

  it("doesn't print excluded fields", () => {
    mockRandom();

    const wrapper = shallow(<DetailObject name={name} detail={detail} exclude={['port']} />);

    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    expect(wrapper.html()).toContain('<span class="text-capitalize">[host]</span>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[subset]</span>');
    expect(wrapper.html()).toContain('<span class="text-capitalize">[weight]</span>');

    expect(wrapper.html()).not.toContain('<strong class="text-capitalize">port</strong>');
    expect(wrapper.html()).not.toContain('<span class="text-capitalize">[number]</span>');
    expect(wrapper.html()).not.toContain('<span class="text-capitalize">[name]</span>');
  });

  it('prints an alert message', () => {
    const validation = {
      message: 'Not all checks passed',
      icon: 'error-circle-o',
      color: PfColors.Red400
    };

    mockRandom();

    const wrapper = shallow(<DetailObject name={name} detail={detail} validation={validation} />);

    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    const iconWrapper = wrapper.find(Icon);
    expect(iconWrapper.prop('type')).toEqual('pf');
    expect(iconWrapper.prop('name')).toEqual(validation.icon);
    expect(iconWrapper.parent().html()).toContain(validation.message);
  });

  it("doesn't print any alert message", () => {
    const validation = {
      message: '',
      icon: 'error-circle-o',
      color: PfColors.Red400
    };

    mockRandom();

    const wrapper = shallow(<DetailObject name={name} detail={detail} validation={validation} />);

    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    const iconWrapper = wrapper.find(Icon);
    expect(iconWrapper.length).toEqual(0);
  });
});
