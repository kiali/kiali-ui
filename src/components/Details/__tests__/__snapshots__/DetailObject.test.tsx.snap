// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailObject test doesn't print any alert message 1`] = `
<div>
  <div>
    <strong
      className="text-capitalize"
    >
      nodejs
    </strong>
    <ul
      className="details"
    >
      <li
        key="key_sssssssss_k0"
      >
        <div>
          <strong
            className="text-capitalize"
          >
            destination
          </strong>
          <ul
            className="details"
          >
            <li
              key="key_sssssssss_k0"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [host]
                  </span>
                   
                  reviews
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k1"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [subset]
                  </span>
                   
                  v1
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k2"
            >
              <div>
                <strong
                  className="text-capitalize"
                >
                  port
                </strong>
                <ul
                  className="details"
                >
                  <li
                    key="key_sssssssss_k0"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [number]
                        </span>
                         
                        22
                      </span>
                    </div>
                  </li>
                  <li
                    key="key_sssssssss_k1"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [name]
                        </span>
                         
                        ssh
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </li>
      <li
        key="key_sssssssss_k1"
      >
        <div
          className="label-collection"
        >
          <span>
            <span
              className="text-capitalize"
            >
              [weight]
            </span>
             
            85
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`DetailObject test doesn't print excluded fields 1`] = `
<div>
  <div>
    <strong
      className="text-capitalize"
    >
      nodejs
    </strong>
    <ul
      className="details"
    >
      <li
        key="key_sssssssss_k0"
      >
        <div>
          <strong
            className="text-capitalize"
          >
            destination
          </strong>
          <ul
            className="details"
          >
            <li
              key="key_sssssssss_k0"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [host]
                  </span>
                   
                  reviews
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k1"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [subset]
                  </span>
                   
                  v1
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k2"
            />
          </ul>
        </div>
      </li>
      <li
        key="key_sssssssss_k1"
      >
        <div
          className="label-collection"
        >
          <span>
            <span
              className="text-capitalize"
            >
              [weight]
            </span>
             
            85
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`DetailObject test prints a nested list with all attributes in the detail 1`] = `
<div>
  <div>
    <strong
      className="text-capitalize"
    >
      nodejs
    </strong>
    <ul
      className="details"
    >
      <li
        key="key_sssssssss_k0"
      >
        <div>
          <strong
            className="text-capitalize"
          >
            destination
          </strong>
          <ul
            className="details"
          >
            <li
              key="key_sssssssss_k0"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [host]
                  </span>
                   
                  reviews
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k1"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [subset]
                  </span>
                   
                  v1
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k2"
            >
              <div>
                <strong
                  className="text-capitalize"
                >
                  port
                </strong>
                <ul
                  className="details"
                >
                  <li
                    key="key_sssssssss_k0"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [number]
                        </span>
                         
                        22
                      </span>
                    </div>
                  </li>
                  <li
                    key="key_sssssssss_k1"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [name]
                        </span>
                         
                        ssh
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </li>
      <li
        key="key_sssssssss_k1"
      >
        <div
          className="label-collection"
        >
          <span>
            <span
              className="text-capitalize"
            >
              [weight]
            </span>
             
            85
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`DetailObject test prints an alert message 1`] = `
<div>
  <div>
    <strong
      className="text-capitalize"
    >
      nodejs
    </strong>
    <div>
      <p
        style={
          Object {
            "color": "#470000",
          }
        }
      >
        <Icon
          name="error-circle-o"
          type="pf"
        />
         
        Not all checks passed
      </p>
    </div>
    <ul
      className="details"
    >
      <li
        key="key_sssssssss_k0"
      >
        <div>
          <strong
            className="text-capitalize"
          >
            destination
          </strong>
          <ul
            className="details"
          >
            <li
              key="key_sssssssss_k0"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [host]
                  </span>
                   
                  reviews
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k1"
            >
              <div
                className="label-collection"
              >
                <span>
                  <span
                    className="text-capitalize"
                  >
                    [subset]
                  </span>
                   
                  v1
                </span>
              </div>
            </li>
            <li
              key="key_sssssssss_k2"
            >
              <div>
                <strong
                  className="text-capitalize"
                >
                  port
                </strong>
                <ul
                  className="details"
                >
                  <li
                    key="key_sssssssss_k0"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [number]
                        </span>
                         
                        22
                      </span>
                    </div>
                  </li>
                  <li
                    key="key_sssssssss_k1"
                  >
                    <div
                      className="label-collection"
                    >
                      <span>
                        <span
                          className="text-capitalize"
                        >
                          [name]
                        </span>
                         
                        ssh
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </li>
      <li
        key="key_sssssssss_k1"
      >
        <div
          className="label-collection"
        >
          <span>
            <span
              className="text-capitalize"
            >
              [weight]
            </span>
             
            85
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
